movies/presentation/src/main/java/com/example/movies/presentation/ui/moviesList/MoviesListView.kt

```diff
@@ -0,0 +1,172 @@
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavDirections
+import coil.compose.rememberAsyncImagePainter
+import coil.request.ImageRequest
+import coil.size.Scale
+import com.example.androidHelpers.compose.views.MovieLottieAnimation
+import com.example.movies.domain.entities.Movie
+import com.example.movies.presentation.R
+
+
+@Composable
+fun MovieList(
+    movieList: List<Movie>?,
+    isLoadingVisible: Boolean = false,
+    onNavigate: (NavDirections) -> Unit,
+    viewModel: MoviesListViewModel
+) {
+    val listState = rememberLazyListState()
+    Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+        if (isLoadingVisible) {
+            MovieLottieAnimation()
+        } else {
+            movieList?.let {
+                LazyColumn(state = listState) {
+                    itemsIndexed(items = movieList) { _, item ->
+                        if (item.id > 0) {
+                            Box(modifier = Modifier
+                                .clickable {
+                                    val navDirection =
+                                        MoviesListFragmentDirections.actionMoviesListFragmentToMovieDetailFragment(
+                                            movieId = item.id
+                                        )
+                                    onNavigate(navDirection)
+                                }
+                            ) {
+                                MovieItem(item)
+                            }
+                        } else if (item.id == 0) {
+                            Box(
+                                modifier = Modifier
+                                    .fillMaxSize(),
+                                contentAlignment = Alignment.Center
+                            ) {
+                                Column() {
+                                    CircularProgressIndicator()
+                                    Text(text = "Loading...")
+                                }
+                            }
+                        }
+                    }
+                }
+                listState.OnBottomReached(buffer = 1) {
+                    viewModel.getMoviesList()
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun MovieItem(movie: Movie?) {
+    Card(
+        modifier = Modifier
+            .padding(8.dp, 4.dp)
+            .fillMaxWidth()
+            .height(100.dp),
+        shape = RoundedCornerShape(8.dp),
+        elevation = 4.dp
+    )
+    {
+        Surface {
+            Row(
+                Modifier
+                    .padding(4.dp)
```

Comment:
Since 4dp is a recurring value maybe it should be extracted to a dimens file

Reply:
Added as a dimen resource